e2e/scenario/scenario_gpu-azurecni.go

```diff
@@ -0,0 +1,33 @@
+package scenario
+
+import (
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+)
+
+func gpu_azurecni() *Scenario {
+	return &Scenario{
+		Name:        "gpu-azurecni",
+		Description: "gpu scenario on cluster configured with Azure CNI",
+		Config: Config{
+			ClusterSelector: NetworkPluginAzureSelector,
+			ClusterMutator:  NetworkPluginAzureMutator,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin = string(armcontainerservice.NetworkPluginAzure)
+				nbc.AgentPoolProfile.KubernetesConfig.NetworkPlugin = string(armcontainerservice.NetworkPluginAzure)
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].VMSize = "Standard_NC6"
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-ubuntu-containerd-18.04-gen2"
+				nbc.AgentPoolProfile.VMSize = "Standard_NC6"
```

Comment:
nit - mismatched with actual VM size

Reply:
looks like we also had this problem for the vanilla gpu scenario :D